.github/workflows/UnitTest.yml

```
@@ -1,31 +1,47 @@
-name: Unit test
-
+name: UnitTest
 on:
   create:
     tags:
   push:
     branches:
       - master
   pull_request:
-
 jobs:
   test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
-        julia-version: ['1.0', '1', 'nightly']
-        os: [ubuntu-latest, windows-latest, macOS-latest]
-
+        version:
+          - '1.0'
+          - '1'
+          - 'nightly'
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macOS-latest
+        arch:
+          - x64
```

macOS (10x) and Windows (3x) resources take more credits than Ubuntu resources, so my idea is to only test macOS and Windows on one Julia version. In most platform-agnositic packages, this should be enough to catch potential issues.

```suggestion
      matrix:
        julia-version: ['1.0', '1', 'nightly']
        os: [ubuntu-latest]
        arch: [x64]
        include:
          - os: windows-latest
            julia-version: '1'
            arch: x64
          - os: macOS-latest
            julia-version: '1'
            arch: x64
          - os: ubuntu-latest
            julia-version: '1'
            arch: x86
```

---

I think that public repos are free regardless, but it doesn't hurt to be nice to them.

From the page you linked:
> GitHub Actions usage is free for both public repositories and self-hosted runners.

---

Oh right. My eyes were caught by the tables and missed the first sentence 😅

---

Still doesn't hurt. I'm sure those numbers measure scarcity, and waste is never good even when you're not paying. So :+1: